Python/bytecodes.c

```diff
@@ -2253,14 +2253,22 @@ dummy_func(
             goto resume_frame;
         }
 
-        inst(POP_JUMP_IF_FALSE, (cond -- )) {
+        inst(POP_JUMP_IF_FALSE, (unused/1, cond -- )) {
             assert(PyBool_Check(cond));
-            JUMPBY(oparg * Py_IsFalse(cond));
+            int flag = Py_IsFalse(cond);
+            #if ENABLE_SPECIALIZATION
+            next_instr->cache = (next_instr->cache << 1) | flag;
+            #endif
+            JUMPBY(oparg * flag);
         }
 
-        inst(POP_JUMP_IF_TRUE, (cond -- )) {
+        inst(POP_JUMP_IF_TRUE, (unused/1, cond -- )) {
             assert(PyBool_Check(cond));
-            JUMPBY(oparg * Py_IsTrue(cond));
+            int flag = Py_IsTrue(cond);
+            #if ENABLE_SPECIALIZATION
+            next_instr->cache = (next_instr->cache << 1) | flag;
+            #endif
+            JUMPBY(oparg * flag);
```

Comment:
Don't you need also a SKIP_OVER the cache?

I'm guessing that could cause the ``assert(frame->prev_instr == instr);`` in ``_Py_call_instrumentation_jump`` to fail for the instrumented jumps.

Reply:
That skip over the cache is already generated -- see the corresponding code in generated_cases.c.h.

Reply:
You mean the "next_instr += 1;"?   In all other cases there is an explicit ``SKIP_OVER(INLINE_CACHE_ENTRIES_...)`` in bytecodes.c.

Reply:
Those SKIP_OVER() calls are always followed by a DISPATCH() call (or maybe a goto).

Reply:
I see. Can we make the code generator emit ``SKIP_OVER(X)`` instead of ``next_instr += x;``?

Reply:
We can, though IIRC Mark at some point objected to emitting macros. So I'd rather keep the status quo.

Reply:
@markshannon What is the reason not to emit macros?

A reason to emit them is so that they are implemented in one place, so if their implementation changes you only change there. Do we want to change the code generator (and to remember that we need to) every time the implementation of a macro like SKIP_OVER changes?

Reply:
Honestly I don't expect SKIP_OVER() to ever change. In hand-written code the macro expresses the intent better. But in generated code it just obscures what happens. I had to go to some lengths to change PEEK() and POKE() calls in the generated code to using stack_pointer[x] instead; I don't want to go back. If you still disagree, try engaging @markshannon.

Reply:
> If you still disagree,

More like trying to understand than disagreeing.

> try engaging @markshannon.

Yes I directed my previous comment to him.